core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -542,6 +553,7 @@ private int provisionInitialProcesses() {
     private int sendOrProcessProvisionedRequests() {
         var processes = transferProcessStore.nextForState(PROVISIONED.code(), batchSize);
         for (TransferProcess process : processes) {
+            telemetry.setCurrentTraceContext(process);
```

Comment:
it seems the context leaks out of the for loop. Use try-with-resources?

Reply:
Alright you have already done it. Good job!